gfx/include/public/gfx/gfx.h

```diff
@@ -2,17 +2,74 @@
 
 #pragma once
 
+#include <gfx/gfx_enums.h>
+#include <gfx/gfx_structs.h>
+#include <gfx/gfx_typedefs.h>
+
+#include <cstdint>
+
 #define GFX_API [[nodiscard]]
 
+#ifndef IMPL
+#define IMPL base
+#endif
+
 namespace antioch::gfx {
-enum class Result {
-  Success,
-  InvalidParameter,
-  OutOfMemory
 
+namespace IMPL {
```

Comment:
I know why you're doing this, but personally I'm not a fan of this style of polymorphism. I would prefer you declare this an abstract class and declare subclasses. then in the cmake you can use different impl files for different target archs

Reply:
I think I could probably just remove the IMPL namespace all together, I'm just gonna need to add more casting to the implementation functions.